packages/loopring_v3/test/testExchangeRingSettlement.ts

```diff
@@ -43,6 +43,52 @@ contract("Exchange", (accounts: string[]) => {
   describe("Trade", function() {
     this.timeout(0);
 
+    it("AMM", async () => {
+      const ring: SpotTrade = {
+        orderA: {
+          owner: exchangeTestUtil.testContext.orderOwners[0],
+          tokenS: "WETH",
+          tokenB: "GTO",
+          amountS: new BN(web3.utils.toWei("99", "ether")),
+          amountB: new BN(web3.utils.toWei("200", "ether")),
+          balanceS: new BN(web3.utils.toWei("10000", "ether")),
+          balanceB: new BN(web3.utils.toWei("20000", "ether")),
+          feeBips: 0,
+          amm: true
+        },
+        orderB: {
+          tokenS: "GTO",
```

Comment:
so orderB represents an AMM pool? Where is the account ID/owner?

Reply:
Is it possible to match two AMM pools?

Reply:
> so orderB represents an AMM pool? 

OrderA is the AMM order (+ initial balances) that can be used without a signature of the order owner, orderB is the taker of the AMM order.

> Where is the account ID/owner?

To make writing tests easier these don't need to be filled in if they just need default values, `setupRing` will fill in the missing data.

> Is it possible to match two AMM pools?

Yes, AMM can currently be enabled for both accounts at the same time.